src/game_objects/process.py

```diff
@@ -135,36 +144,41 @@ def _on_io_event(self):
 
     def _terminate_gracefully(self):
         if self._process_manager.terminate_process(self, False):
+            event_manager.event_process_terminated(self._pid)
             self._has_ended = True
             self._set_waiting_for_io(False)
             self._set_waiting_for_page(False)
             self._starvation_level = 0
 
     def _terminate_by_user(self):
         if self._process_manager.terminate_process(self, True):
+            event_manager.event_process_killed(self._pid)
             self._has_ended = True
             self._set_waiting_for_io(False)
             self._set_waiting_for_page(False)
             self._starvation_level = 6
             for page in self._pages:
+                event_manager.event_page_free(page.pid, page.idx)
                 self._page_manager.delete_page(page)
+            del Process.Processes[self._pid]
 
     def _check_if_clicked_on(self, event):
         if event.type == GameEventType.MOUSE_LEFT_CLICK:
             return self.starvation_level < 6 and self._view.collides(
                 *event.get_property('position'))
         return False
 
-    def _on_click(self):
+    def on_click(self):
```

Comment:
Same here, would create an intermediary method and make that one public instead. Maybe `toggle`, but I'm open to better ideas.